aopy/analysis/base.py

```diff
@@ -441,27 +441,44 @@ def calc_sem(data, axis=None):
 
     return SEM
 
-def calc_rolling_average(data, window_size=11):
+def calc_rolling_average(data, window_size=11, mode='copy'):
     """
-    Computes the rolling average of a 1D array using a convolutional kernel. The
-    first and last valid datapoint (fully overlapping with the kernel) are copied
-    backwards and forwards, respectively, such that the size of the output is the
-    same as the size of the input.
+    Computes the rolling average of a 1- or 2-D array using a convolutional kernel. 
+    The rolling average is always applied along the first axis of the array.
```

Comment:
This should work for an n-D array right?

Reply:
Yeah probably, but untested